source/extensions/tracers/opentelemetry/resource_detectors/dynatrace/dynatrace_metadata_file_reader.cc

```diff
@@ -0,0 +1,56 @@
+#include "source/extensions/tracers/opentelemetry/resource_detectors/dynatrace/dynatrace_metadata_file_reader.h"
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "envoy/common/exception.h"
+
+#include "source/common/common/logger.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+
+namespace {
+
+bool isIndirectionFile(const std::string& file_name) {
+  return absl::StartsWith(file_name, "dt_metadata_");
+}
+
+const std::string readFile(const std::string& file_name) {
+  if (file_name.empty()) {
+    return "";
+  }
+
+  std::ifstream file(file_name);
```

Comment:
use `fileReadToEnd`?

Reply:
As I explained in the issue/pr description, the files that we read are "virtual", they don't exist in the file system. The Dynatrace OneAgent instrumentation intercepts the file read call and then provide the file on demand. 

If I use `fileReadToEnd` it ends up calling this https://github.com/envoyproxy/envoy/blob/d6dc92f64f62b0f597b2a92bfcebc7af2dbc0da5/source/common/filesystem/posix/filesystem_impl.cc#L346 which fails always because the file does not exist. That's why I didn't use the file system abstractions in Envoy.